### PR TITLE
CP-18300: Allow choosing PVS-cache VDI size per SR

### PR DESCRIPTION
**Pull request by euanh**

### CODE REVIEW

File: ocaml/xapi/cli_frontend.ml [hunk 1]
```diff
@@ -2675,6 +2675,22 @@ add a mapping of 'path' -> '/tmp', the command line should contain the argument
 			implementation=No_fd Cli_operations.PVS_proxy.destroy;
 			flags=[];
 		};
+		"pvs-cache-storage-create",
+		{
+			reqd=["pvs-site-uuid"; "sr-uuid"; "size"];
+			optn=[];
+			help="Configure a PVS cache on a given SR";
+			implementation=No_fd Cli_operations.PVS_proxy.create;
```
Comment (robhoes) [inline ocaml/xapi/cli_frontend.ml]: Wrong function here. This could be why you got Not_found errors.

**Comment by robhoes** [comment]
This looks mostly right. There is a bit missing in the destroy function: compare it with the original one and just make it do the same. Also please add a handler for the new class to db_gc.ml as a "connector" between `SR` and `PVS_site`.

**Comment by robhoes** [comment]
- "CP-18300: Add constraints to PVS_cache_storage create and destroy" only touches `create`.
- `destroy` does not yet call the cleanup function `Xapi_pvs_cache.on_sr_remove ~__context ~sr:value`.
- The VDI size is still hardcoded.

**Comment by euanh** [comment]
Gah, the call to `Xapi_pvs_cache.on_sr_remove` fell off while rebasing. :(

**Comment by robhoes** [comment]
OK, just the size then :)

**Pull request merged**